components/brave_wallet_ui/common/slices/api.slice.ts

```diff
@@ -370,12 +374,15 @@ export function createWalletApi (
       getDefaultAccountAddresses: query<string[], void>({
         queryFn: async (arg, { dispatch }, extraOptions, baseQuery) => {
           // apiProxy
-          const { keyringService, jsonRpcService } = baseQuery(undefined).data
+          const { braveWalletService, keyringService, jsonRpcService }
+            = baseQuery(undefined).data
 
           // Get default account addresses for each CoinType
           const defaultAccountAddresses = await Promise.all(
             SupportedCoinTypes.map(async (coin: BraveWallet.CoinType) => {
-              const { chainId } = await jsonRpcService.getChainId(coin)
+              const { originInfo } = await braveWalletService.getActiveOrigin()
+              const { chainId }
+                = await jsonRpcService.getChainId(coin, originInfo.origin)
```

Comment:
This could be `jsonRpcService.getChainIdForActiveOrigin(coin)`, so jsonRpcService could be itself responsible for determining what is current active origin(through some delegate). 
I don't think we actually need arbitrary origin support at frontend and it is always active origin

Reply:
I will export that new interface in `BraveWalletService` since it already has delegate fetching active origin and also access to raw pointer of JsonRpcService which should be decoupled from determining current active origin.

Reply:
fixed in 210bbec4829bea9ca6ae39e73d2bbd6bb233ad9e and 4ebbe1249cb16e6694ddefe20b5c7505a265bcc3